src/biopsykit/classification/model_selection/sklearn_pipeline_permuter.py

```diff
@@ -125,34 +136,66 @@ def __init__(
         """List of model combinations, i.e. permutations of the different transformers/estimators for
         each pipeline step."""
 
-        self.grid_searches: Dict[Tuple[str, str], pd.DataFrame] = {}
-        """Dictionary with grid search search results for each pipeline step combination."""
+        self.hyper_search_dict: Dict[str, Dict[str, Any]] = {}
+        """Dictionary specifying the selected hyperparameter search method for each estimator."""
+
+        self.param_searches: Dict[Tuple[str, str], pd.DataFrame] = {}
+        """Dictionary with parameter search results for each pipeline step combination."""
 
         self.results: Optional[pd.DataFrame] = None
         """Dataframe with parameter search results of each pipeline step combination."""
 
         self.scoring: str = ""
+        """Scoring used as metric for optimization during hyperparameter search."""
+
+        self._results_set: bool = False
 
         if kwargs.get("score_summary") is not None:
             self.results = kwargs.get("score_summary")
-        else:
-            for category in model_dict:
-                if not set(model_dict[category].keys()).issubset(set(param_dict.keys())):
-                    missing_params = list(set(model_dict[category].keys()) - set(param_dict.keys()))
-                    raise ValueError("Some estimators are missing parameters: {}".format(missing_params))
+            return
 
-            model_combinations = list(
-                product(*[[(step, k) for k in list(model_dict[step].keys())] for step in model_dict])
-            )
+        self._check_missing_params(model_dict, param_dict)
+
+        if hyper_search_dict is None:
+            hyper_search_dict = {}
+        self.hyper_search_dict = hyper_search_dict.copy()
+
+        clf_list = model_dict[list(model_dict.keys())[-1]]
+        for clf in clf_list:
+            self.hyper_search_dict.setdefault(clf, {"search_method": "grid"})
```

Comment:
maybe just one inline comment to explain that the dict is filled up with the classifiers that are not specified in `self.hyper_search_dict` with "grid"